packages/client-authentication/index.js

```diff
@@ -0,0 +1,23 @@
+'use strict'
+
+const debug = require('debug')('xmpp:client:authentication')
+
+function authenticate (creds, cb) {
+  const auth = this.authenticators.find(auth => auth.match(this.features))
```

Comment:
Arrow functions are only supported on node 4 and up. Are we dropping support for 0.12?